### DISCUSSION TITLE
Boolean based on workflow conditions

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#27029] [Actions]
<p>How would you go about creating a boolean variable, which switches its value based on certain conditions ?</p>
<p>The variable is used inside a script in the workflow.</p>
<p>I’ve came up with this solution through Powershell:</p>
<pre><code class="lang-auto">      - name: Determine if signing should be applied to AppxBundle
        id: determine_signing
        shell: powershell
        run: |
          [bool]$PARAM = $false
          if ( ($env:GITHUB_EVENT_NAME -ceq 'push') -and ($env:GITHUB_REF -ceq 'refs/heads/master') -and ($env:BUILD_CONFIGURATION -ceq 'Release') ) {
            $PARAM = $true
          }
          echo "::set-output name=parameter::$(echo $PARAM)"
        env:
          $GITHUB_EVENT_NAME: ${{ github.event_name }}
          $GITHUB_REF: ${{ github.ref }}
          $BUILD_CONFIGURATION: ${{ matrix.configuration }}
</code></pre>
<p>And of course it works, but I was thinking about a solution that is YAML only. The thing is, I can’t think of any :smiley:</p>
<p>Any ideas ?</p>

**Comment by Simran-B** [comment]
<p>You mean like this?</p>
<pre><code class="lang-yaml">- name: Run on push to master if build config is Release
  if: github.event_name == 'push' &amp;&amp; github.ref == 'refs/heads/master' &amp;&amp; env.BUILD_CONFIGURATION == 'Release'
  run: |
    echo "The conditions were met"
</code></pre>
<p><code>github</code> context: <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" class="inline-onebox" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a><br>
<code>env</code> context: <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context" class="inline-onebox" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a></p>
<p><code>env.BUILD_CONFIGURATION</code> will probably only be available if you set it via <code>env: ...</code> in the workflow itself, not if you set it in a <code>run: ...</code> block like <code>export BUILD_CONFIGURATION="Release"</code>. You can set an env key from within a <code>run</code> block with a workflow command if necessary, however, e.g. like <code>echo "BUILD_CONFIGURATION=Release" &gt;&gt; $GITHUB_ENV</code> using bash shell (also see <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" class="inline-onebox" rel="noopener nofollow ugc">Workflow commands for GitHub Actions - GitHub Docs</a>).</p>

**Comment by aleks-ivanov** [comment]
<p>It was quite some time ago, but I think I meant grouping conditions inside a single variable.</p>
<pre><code class="lang-auto">env:
  condition: github.event_name == 'push' &amp;&amp; github.ref == 'refs/heads/master' &amp;&amp; env.BUILD_CONFIGURATION == 'Release'
</code></pre>
<p>Several months later, I don’t think that’s possible in YAML, but correct me if I am wrong.</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Well, you could evaluate an expression in place, like this:</p>
<pre><code class="lang-auto">env:
  condition: ${{ github.event_name == 'push' &amp;&amp; github.ref == 'refs/heads/master' &amp;&amp; env.BUILD_CONFIGURATION == 'Release' }}
</code></pre>
<p>But the <code>env</code> context is only available in steps, so the third condition wouldn’t be possible like this.</p>
<p>Or do you mean to defer evaluation? As in have <code>github.event_name == 'push' &amp;&amp; github.ref == 'refs/heads/master' &amp;&amp; env.BUILD_CONFIGURATION == 'Release'</code> as string and then evaluate that string in certain places of the workflow, something along <code>${{ eval(env.condition) }}</code>? That is definitely not supported, and I would guess it’s very unlikely to ever get supported because of security concerns.</p>

**Comment by aleks-ivanov** [comment]
<p>That’s it!</p>
<p>Thank you!</p>
